.github/workflows/ci.yml

```
@@ -0,0 +1,50 @@
+on: [push, pull_request]
+name: build
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        ghc: ['8.8.4','8.10.7']
+        os: [ubuntu-latest]
+    name: Haskell GHC ${{ matrix.ghc }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: haskell/actions/setup@v1.2.9
+        id: setup-haskell-build
+        with:
+          ghc-version: ${{ matrix.ghc }} # Resolves to the latest point release of GHC 8.8
+          cabal-version: '3.6.2.0' # Exact version of Cabal
+          enable-stack: true
+          stack-version: '2.7'
+      - name: Install dependencies
+        run: sudo apt-get install -y libxml2-utils libgl1-mesa-dev libglu1-mesa-dev freeglut3-dev libpulse-dev libblas-dev liblapack-dev libasound2-dev
+      - name: Cabal update
+        run: cabal update
+      - name: Freeze Cabal plan
+        run: cabal freeze
+      - name: Cache Cabal build artifacts
+        uses: actions/cache@v2
+        with:
+          path: |
+            ${{ steps.setup-haskell-build.outputs.cabal-store }}
+            dist-newstyle
+          key: ${{ runner.os }}-cabal-${{ matrix.ghc }}-${{ hashFiles('cabal.project.freeze') }}
+          restore-keys: |
+            ${{ runner.os }}-cabal-${{ matrix.ghc }}
+      - name: Cabal build dependencies
+        run: |
+          cabal build all --enable-tests --only-dependencies
+      - name: Cabal build all
+        run: |
+          cabal build all --enable-tests
+      - name: Cabal test
+        run: |
+          cabal test all --enable-tests --test-show-details=Always
+      - name: Cache stack build artifacts
+        uses:  freckle/stack-cache-action@v1.0.1 
+        with:
+          stack-yaml: stack.yaml
+      - name: Stack build and test (lts-18.21 - ghc 8.10.7)
```

```suggestion
      - name: Stack build and test (lts-18.21)
```

(GHC version is part of the matrix)